models/articles-model.js

```diff
@@ -1,11 +1,39 @@
 const db = require("../db/connection")
 
-exports.fetchArticleById = async (article_id) => {
+const fetchArticleById = async (article_id) => {
     const query = `SELECT * FROM articles WHERE article_id = $1`   
     const body = await db.query(query, [article_id])
     if (body.rows.length === 0) {
         return Promise.reject({ status: 404, message: "article id not found"})
     } else {
-        return body.rows
+        return body.rows[0]
     }        
-}
\ No newline at end of file
+}
+
+
+
+
+
+
+
+
+
+const fetchArticleComments = async (article_id) => {
+    const validArticle = await fetchArticleById(article_id)
+    if (validArticle.length === 0) {
+        return Promise.reject({ status: 404, message: "article id not found"})
+    }
+    
+    const query = `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`
+    const body = await db.query(query, [article_id])
+
+    if (body.rows.length === 0) {
+        return Promise.reject({ status: 200, message: "no comments found for this article"})
```

Comment:
It's arguable that you could just return rows here without the if statement. An error is usually for when the client or server has done something wrong, however everything has gone according to plan here, there simply aren't any comments, sending back an empty array here would also be sufficient :)